packages/create-amplify/src/initial_project_file_generator.ts

```diff
@@ -52,6 +69,20 @@ export class InitialProjectFileGenerator {
       'es2022',
     ];
 
-    await this.executeWithDebugLogger(targetDir, 'npx', tscArgs, execa);
+    if (this.executableName.startsWith('yarn')) {
+      await this.executeWithDebugLogger(
+        targetDir,
+        'yarn',
+        ['add', 'typescript@^5'],
+        execa
+      );
```

Comment:
why do we need this ?

Reply:
Before we convert to hybrid PnP and Node_modules, we had to install typescript in `amplify/` folder to make sure `tsc` works, but since we adopt the hybrid mode, I can try if we still need to install typescript.

Reply:
please check.

Reply:
Just checked and we have to install typescript at both root and `amplify/` folder because 
1) `amplify/` folder doesn't have access to the modoles in root folder if it's not a workspace
2) we use `tsc` from the root folder in the e2e test for compile. Also we may use `tsc` for deploy.